src/fabric/src/fabric_doc_open.erl

```diff
@@ -140,13 +139,22 @@ read_repair(#acc{dbname = DbName, replies = Replies, node_revs = NodeRevs}) ->
             choose_reply(Docs);
         [#doc{id = Id} | _] ->
             Opts = [?ADMIN_CTX, ?REPLICATED_CHANGES, {read_repair, NodeRevs}],
-            Res = fabric:update_docs(DbName, Docs, Opts),
-            case Res of
-                {ok, []} ->
-                    couch_stats:increment_counter([fabric, read_repairs, success]);
-                _ ->
-                    couch_stats:increment_counter([fabric, read_repairs, failure]),
-                    couch_log:notice("read_repair ~s ~s ~p", [DbName, Id, Res])
+            case fabric_util:is_replicator_db(DbName) of
+                true ->
+                    % Skip read_repair for _replicator shards. We might have
+                    % gotten the result of after_doc_read/2 with a fake patched
+                    % revision and stripped creds. Let the internal replicator
+                    % handle all that
+                    ok;
```

Comment:
```suggestion
                    couch_log:debug("~p skipping read repair for ~s", [?MODULE, DbName]),
                    ok;
```

Reply:
If it weren't for the timer:sleep/1 in the test we wouldn't have wanted to add a log line there to start with, so I think an extra 100 msec is not that bad in that case just to keep the main code a bit cleaner.